### PR TITLE
`Programming exercises`: Improve layout of taskbar

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).
#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

The task bar uses two table cells. One for "Task"/"Teilaufgaben" and the other for all the task icons. We want to support up to 15 task icons, so they should get the majority of space compared to the one word at the start of the bar.

### Description

This PR shrinks the cell of the one word to its minimum size to give the task icons more of the available space. This especially reduces the unused space between the one word and the first task icon.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
- 1 Programming Exercise

1. Log in to Artemis
2. Create the exercise/Find one with some tasks in the problem statement
3. Look at the task bar to see the icons that are more evenly spaced out

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Before:
Bildschirmfoto 2023-06-18 um 16 09 51
Bildschirmfoto 2023-06-18 um 16 10 00

With this PR:
Bildschirmfoto 2023-06-18 um 16 09 38
Bildschirmfoto 2023-06-18 um 16 10 13

### CODE REVIEW

File: src/main/webapp/app/exercises/programming/shared/instructions-render/step-wizard/programming-exercise-instruction-step-wizard.scss [hunk 1]
```diff
@@ -4,12 +4,6 @@
     border-bottom: 1px solid var(--programming-exercise-instruction-step-wizard-card-header-border);
 }
 
-.stepwizard-row {
-    display: flex;
-    flex-wrap: wrap;
-    justify-content: space-evenly;
-}
-
```
Comment (JohannesStoehr) [inline src/main/webapp/app/exercises/programming/shared/instructions-render/step-wizard/programming-exercise-instruction-step-wizard.scss:12]: This was only moved down to better reflect the hierarchy/order in the html template

**Review by SolizerCodes — APPROVED** [review]
Looks good on TS6 👍

**Review by DominikRemo — APPROVED** [review]
Tested on ts1. Looks good

**Review by RY997 — APPROVED** [review]
Tested on ts4, worked well

**Review by eceeeren — APPROVED** [review]
Tested on TS2, works well

**Review by greekgeek1 — APPROVED** [review]
Tested on ts1. Works as expected.

**Review by MichaelOwenDyer — APPROVED** [review]

**Review by tobias-lippert — APPROVED** [review]
Code LGTM

**Pull request merged**